cmd/api_example/main.cpp

```diff
@@ -70,74 +43,69 @@ verify(const std::string& label, Session& alice, Session& bob)
 int
 main() // NOLINT(bugprone-exception-escape)
 {
+  const auto suite =
+    mls::CipherSuite{ mls::CipherSuite::ID::X25519_AES128GCM_SHA256_Ed25519 };
+
   ////////// DRAMATIS PERSONAE ///////////
 
-  auto alice = User{ "alice" };
-  auto bob = User{ "bob" };
-  auto charlie = User{ "charlie" };
+  auto alice_client = create_client(suite, "alice");
+  auto bob_client = create_client(suite, "bob");
+  auto charlie_client = create_client(suite, "charlie");
 
   ////////// ACT I: CREATION ///////////
 
-  // Bob posts a KeyPackage
-  auto kpB = bob.fresh_key_package();
-
-  // Alice starts a session with Bob
-  auto infoA = alice.temp_init_info();
   auto group_id = bytes{ 0, 1, 2, 3 };
-  auto [sessionA, welcome] =
-    Session::start(group_id, { infoA }, { kpB }, random_bytes(32));
-
-  // Bob looks up his CIK based on the welcome, and initializes
-  // his session
-  auto sessionB = Session::join(bob.infos(), welcome);
-
-  // Alice and Bob should now be on the same page
-  verify("create", sessionA, sessionB);
+  auto alice_session = alice_client.begin_session(group_id);
 
   ////////// ACT II: ADDITION ///////////
 
-  // Charlie posts a KeyPackage
-  auto kpC1 = charlie.fresh_key_package();
-
-  // Alice adds Charlie to the session
-  bytes add;
-  std::tie(welcome, add) = sessionA.add(random_bytes(32), kpC1);
+  // Bob and Charlie post KeyPackages
+  auto bob_join = bob_client.start_join();
+  auto charlie_join = charlie_client.start_join();
 
```

Comment:
not that it is needed , but as much as i think auto is a good thing, knowing the type here will help readability without needing to look at the source code .